VimeoUpload/Upload/Networking/New Upload (Private)/VimeoSessionManager+Upload.swift

```diff
@@ -29,9 +29,17 @@ import VimeoNetworking
 
 extension VimeoSessionManager
 {
-    func createVideoDataTask(url: URL, videoSettings: VideoSettings?, completionHandler: @escaping VideoCompletionHandler) throws -> URLSessionDataTask
+    func createVideoDataTask(url: URL, videoSettings: VideoSettings?, uploadApproach: VIMUpload.UploadApproach?, completionHandler: @escaping VideoCompletionHandler) throws -> URLSessionDataTask
     {
-        let request = try (self.requestSerializer as! VimeoRequestSerializer).createVideoRequest(with: url, videoSettings: videoSettings)
+        let request: NSMutableURLRequest
+        if let uploadApproach = uploadApproach
+        {
+            request = try (self.requestSerializer as! VimeoRequestSerializer).createVideoRequest(with: url, videoSettings: videoSettings, uploadType: uploadApproach)
```

Comment:
I know we don't really have test coverage in VimeoUpload at the moment, but all of this stuff looks ripe for testing. If not now, it might be worth making a ticket for it.